packages/plugins/flow/resolvers/src/index.ts

```diff
@@ -46,12 +46,21 @@ export const plugin: PluginFunction<RawFlowResolversConfig, Types.ComplexPluginO
     defsToInclude.push(`export type RecursivePick<T, U> = T`);
   }
 
-  const header = `export type Resolver<Result, Parent = {}, Context = {}, Args = {}> = (
+  // ES6 class resolvers methods only contain 3 argument (args, context, info). see https://github.com/apollographql/apollo-server/issues/1996
+  const resolverType = config.supportES6Classes
+    ? `export type Resolver<Result, Parent = {}, Context = {}, Args = {}> = (
+  args: Args,
+  context: Context,
+  info: GraphQLResolveInfo
+) => Promise<Result> | Result;`
+    : `export type Resolver<Result, Parent = {}, Context = {}, Args = {}> = (
```

Comment:
this is a lot for my eyes to process as a single ternary block

I'll gingerly suggest splitting this up, and using a proper if statement? this would add some whitespace to break this up, and may improve the readability

Reply:
Thanks! fixed it, have much less code to read now.